### ISSUE TITLE
Doorbird completely down since 0.1.0

### ISSUE THREAD

**Issue by TheRealGros** [hjdhjd/homebridge-doorbird#69]
Since the update to 0.1.0 my DoorBird is completely frozen. I have to restart it by disconnecting from power. It works well until I access it through Homebridge... any idea how to solve this?

**Comment by brownad** [comment]
Suggest reinstalling the latest DoorBird plugin, also clear out from DoorBird the legacy API settings as they are automatically setup for you now.  There is a possibility something is at play between the plugin to DoorBird, but to be sure, remove the old one as per 

https://github.com/brownad/homebridge-doorbird/issues/71#issuecomment-640407689

You might have to do more surgery on Homebridge e.g. a refresh of the Name / PIN to force an update and install the latest.

**Comment by TheRealGros** [comment]
"also clear out from DoorBird the legacy API settings as they are automatically setup for you now"

Would do this, but how?

**Comment by brownad** [comment]
From your DoorBird app directly, login to App / Admin / API delete the two entries you see… if you had done this before.  If not ignore

**Comment by TheRealGros** [comment]
Thanks, found it (German app, was named different). Upgraded to 0.3.1, deleted the cache, renintitialized Homebridge, seems to work now. Thanks!

**Comment by TheRealGros** [comment]
Called Home on my iPad this morning. Doorbird again freezes completely.
Homebridge log shows "ffmpeg exited with code 1", had to restart DoorBird by unplugging form power.
is there any additional log somewhere to see what is going ob?

**Comment by hjdhjd** [comment]
Turn on ffmpeg debugging in the videoConfig section in your config.json. Your problem is going to be a difficult one to troubleshoot and may not be related to this plugin...it’s difficult to tell, unfortunately. I have seen no reports of issues like this aside from yours. Anecdotally, this sounds like a potential hardware problem...but let’s see what’s going on when you get debugging output from ffmpeg. Have you tweaked any of the defaults for video, or left them alone?

Here are my suggested steps for you:

- turn on debugging in the videoConfig section of config.json.
- ensure you aren’t changing any of the defaults aside from turning on video debugging in the videoConfig section.
- disable audio support.

See what errors are produced. The above is a pretty conservative setup that should work in pretty much all cases.

**Comment by TheRealGros** [comment]
I got a long output, pls. see attached log.

[ffmpeg_debug_output.log](https://github.com/brownad/homebridge-doorbird/files/4752038/ffmpeg_debug_output.log)

**Comment by TheRealGros** [comment]
This is the config I use, pretty simple one:

    {
           "name": "Doorbird",
            "doorbirds": [
                {
                    "name": "Doorbird",
                    "ip": "192.168.188.34",
                    "username": "ggzmmz0002",
                    "password": "xxxxxxxxxx"
                }
            ],
            "videoConfig": {
                "debug": true
            },
            "platform": "Doorbird"
        },
    

And this is a fresh raspberry os install with a fresh Homebridge installation. No tweaks, no changes.

**Comment by hjdhjd** [comment]
Looking at the log file it could be either the version off ffmpeg you're running, or a genuine problem getting a stream from the Doorbird.

You caught an inadvertent bug that's fixed in the Github version...audio support seems to be enabled by default and it shouldn't be. That's fixed now and @brownad will push an update I just committed, hopefully in the next day or so.

Once that's released, here's what I would try, in order and see if any of them help:

- try this again, hopefully this time with properly disabled audio support by default.
- add `"additionalCommandline": "-tune zerolatency"` to the videoConfig block.
- if that doesn't work, try updating the version of ffmpeg you've got to a recent one.

Basically, the first should be a pretty "out of the box' configuration that's safe. The second is a very conservative set of settings to ffmpeg to see if that helps in getting the stream going safely. The third is seeing if there's an inadvertent bug in the version of ffmpeg you're on that you're running into.

If none of that helps...it may be something beyond our ability to easily diagnose I'm afraid.

**Comment by brownad** [comment]
I’ll publish later tonight.. 👍
